src/web/js/output-ui.js

```diff
@@ -550,35 +335,34 @@
         else if(isJSImport(src)) {
           /* NOTE(joe): No special handling here, since it's opaque code */
         }
+        srcElem.on("mouseover", function() {
+          flashMessage("This code is not in this editor.");
+        }).on("mouseleave", clearFlash);
       }
       return srcElem;
     }
-
-    function cmPosFromSrcloc(runtime, srcloc, loc) {
-      return runtime.ffi.cases(runtime.getField(srcloc, "is-Srcloc"), "Srcloc", loc, {
-        "builtin": function(_) {
-           throw new Error("Cannot get CodeMirror loc from builtin location");
-        },
-        "srcloc": function(source, startL, startC, startCh, endL, endC, endCh) {
-          var extraCharForZeroWidthLocs = endCh === startCh ? 1 : 0;
-          return {
-            source: source,
-            start: { line: startL - 1, ch: startC },
-            end: { line: endL - 1, ch: endC + extraCharForZeroWidthLocs }
-          };
-        }
-      });
-    }
-
-    function emphasizeLine(editors, cmloc) {
-      var editor = editors[cmloc.source];
-      if(!editor) return;
-      for(var i = cmloc.start.line; i <= cmloc.end.line; i++) {
-        setTimeout(
-          function(){editor.removeLineClass(this, "background", "emphasize-line");}
-            .bind(editor.addLineClass(i, "background", "emphasize-line")),
-          500);
+    
+    function drawPosition(position) {
```

Comment:
Can `drawPosition` and `drawSrcloc` be harmonized/have their common parts factored out?  They seem very similar.